### ISSUE TITLE
configuration was resolved without accessing the project in a safe manner

### ISSUE THREAD

**Issue by Vampire** [gradle/gradle#10385]
I had a build running with `-t`.
Then I edited one of the Gradle Kotlin scripts in `buildSrc` and this was displayed:
```
Waiting for changes to input files of tasks... (ctrl-d then enter to exit)
The configuration :buildSrc:kotlinScriptDefExtensions was resolved without accessing the project in a safe manner.  This may happen when a configuration is resolved from a thread not managed by Gradle or from a different project.  See https://docs.gradle.org/5.4.1/userguide/troubleshooting_dependency_resolution.html#sub:configuration_resolution_constraints for more details. This behaviour has been deprecated and is scheduled to be removed in Gradle 6.0.
The configuration :buildSrc:kotlinScriptDef was resolved without accessing the project in a safe manner.  This may happen when a configuration is resolved from a thread not managed by Gradle or from a different project.  See https://docs.gradle.org/5.4.1/userguide/troubleshooting_dependency_resolution.html#sub:configuration_resolution_constraints for more details. This behaviour has been deprecated and is scheduled to be removed in Gradle 6.0.
new file: D:\Sourcecode\other\command-framework\buildSrc\src\main\kotlin\net\kautler\tests.gradle.kts
Change detected, executing build...
```

**Comment by bamboo** [comment]
@big-guy, considering the name of the configuration it seems to be coming from the Kotlin scripting plugin:

https://github.com/JetBrains/kotlin/blob/master/libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/scripting/internal/ScriptingGradleSubplugin.kt#L132

**Comment by anumalasri** [comment]
When running a Multi-Project with _--parallel_ option, I am getting below error .. 

Is this related to this Issue ? 

```
The configuration :projectX:compileClasspath was resolved without accessing the project in a safe manner.  This may happen when a configuration is resolved from a thread not managed by Gradle or from a different project.  
See https://docs.gradle.org/5.4.1/userguide/troubleshooting_dependency_resolution.html#sub:configuration_resolution_constraints for more details. This behaviour has been deprecated and is scheduled to be removed in Gradle 6.0
```

**Comment by Vampire** [comment]
This seems to have been fixed between 5.4.1 and 5.6.2

**Comment by denis-ismailaj** [comment]
I'm on 6.8.2 and this happens on Kotlin Multiplatform projects when you enable parallel execution.

**Comment by lestephane** [comment]
Also getting this error on 6.8.2, and 7 (See below). In reference to @denis-ismailaj, `org.gradle.parallel=true` when that happens, but I do not have a multiplatform project. The innards of Gradle don't like a specific part of the domain model to be accessed in parallel it seems. But the warning does not give a clue as to what the problem is or what to do about it.

```
Resolution of the configuration :logging-api:detachedConfiguration1 was attempted from a context different than the project context. Have a look at the documentation to understand why this is a problem and how it can be resolved. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. See https://docs.gradle.org/7.0/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors for more details.
```
How is anybody supposed to fixed this from this (guaranteed to be incomplete from the wording) bullet list:
```
For example:

    A task from one project directly resolves a configuration in another project in the task’s action.

    A task specifies a configuration from another project as an input file collection.

    A build script for one project resolves a configuration in another project during evaluation.

    Project configurations are resolved in the settings file.
```
I have absolutely no clue what Gradle is trying to tell me.

**Comment by Medine17eii** [comment]
> Also getting this error on 6.8.2, and 7 (See below). In reference to @denis-ismailaj, `org.gradle.parallel=true` when that happens, but I do not have a multiplatform project. The innards of Gradle don't like a specific part of the domain model to be accessed in parallel it seems. But the warning does not give a clue as to what the problem is or what to do about it.
> 
> ```
> Resolution of the configuration :logging-api:detachedConfiguration1 was attempted from a context different than the project context. Have a look at the documentation to understand why this is a problem and how it can be resolved. This behaviour has been deprecated and is scheduled to be removed in Gradle 8.0. See https://docs.gradle.org/7.0/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors for more details.
> ```
> 
> How is anybody supposed to fixed this from this (guaranteed to be incomplete from the wording) bullet list:
> 
> ```
> For example:
> 
>     A task from one project directly resolves a configuration in another project in the task’s action.
> 
>     A task specifies a configuration from another project as an input file collection.
> 
>     A build script for one project resolves a configuration in another project during evaluation.
> 
>     Project configurations are resolved in the settings file.
> ```
> 
> I have absolutely no clue what Gradle is trying to tell me.

**Comment by Medine17eii** [comment]
..

**Comment by Medine17eii** [comment]
U

**Comment by Jire** [comment]
Why is this closed.. still entirely not fixed.

**Comment by 3flex** [comment]
In my case it only seems to happen when syncing Gradle with IntelliJ. But when I run `./gradlew help` I don't get the messages.

The messages only relate to configurations that are declared in Kotlin build scripts and only referenced within the same build file i.e. they are not being resolved in any other project or the settings file.

**Comment by jemshit** [comment]
Am i understanding this correctly, `org.gradle.parallel` is introduced to parallel build multiple modules, but it is sin to do so?

**Comment by ov7a** [comment]
Sorry for the late reply.

Your questions might be answered in Gradle documentation or on our community Slack or forums. Find links to these resources on [help.gradle.org](https://help.gradle.org/).

If you think you have come across a Gradle bug, please file a new issue describing the problem and how to reproduce it.  

***

I don't think this is an actionable issue because the reports in the comments are either old or are related to another issue and missing information to reproduce the issue. Please open a new issue if you still experiencing problems.

The original issue was about `:buildSrc:kotlinScriptDef` and  `:buildSrc:kotlinScriptDefExtensions` configurations and it is confirmed to be fixed.

**Comment by Vampire** [comment]
@jemshit I don't think you understood correctly. It is not a sin to use it, but it seems it triggered some bug. And nowadays it is preferable to make a build configuration-cache compatible and enable it, as then all tasks that do not have an ordering constraint can run in parallel, even within one project.
